### PR TITLE
tests: spinlock: add some error test cases

### PR DESCRIPTION
**Pull request by enjiamai**
Add some error test cases for spinlock, include:
1.Validate indentical spinlock cannot be used recursively.
2.Validate unlocking incorrect spinlock will trigger assertion.
3.Validate releasing incorrect spinlock will trigger assertion.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

File: tests/kernel/spinlock/src/spinlock_error_case.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#else
+void assert_post_action(const char *file, unsigned int line)
+#endif
+{
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+	ARG_UNUSED(file);
+	ARG_UNUSED(line);
+#endif
+
+	printk("Caught assert failed\n");
```
Comment (LeiW000) [inline tests/kernel/spinlock/src/spinlock_error_case.c:41]: I guess that it should be "Caught an assert."

**Review by LeiW000 — CHANGES_REQUESTED** [review]
One nit. Otherwise, it looks goog to me.

File: tests/kernel/spinlock/src/spinlock_error_case.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/spinlock/src/spinlock_error_case.c]: Thanks for correcting this, got it fix!

**Review by LeiW000 — APPROVED** [review]

**Review by andyross — CHANGES_REQUESTED** [review]
These depend on CONFIG_SPIN_VALIDATE being true.  You should put that into the prj.conf for this test explicitly.  Right now it's working only because that's true by default on non-tiny platforms.

Beyond that everything looks great.

**Comment by enjiamai** [comment]
@andyross Thanks for your review and instruction. I have set CONFIG_SPIN_VALIDAT=y in prj.conf.  And, we wish you a Merry Christmas!! ^^

**Review by andyross — APPROVED** [review]

**Pull request merged**